Update dbt docs (#33)
glean-io/documentation: pages/docs/data-ops/dbt-integration.mdx

import { Steps } from 'nextra-theme-docs'

# dbt Integration

## Overview

> Glean can import models directly from dbt by specifying metrics in your dbt yaml file. Glean will automatically sync dbt documentation and keep metrics in sync with downstream dashboards (whether they are checked into code or not).  It's also possible to integrate with CI/CD processes.

Reviewer: ```suggestion
Glean can import models directly from dbt by specifying metrics in your dbt yaml file. Glean will automatically sync dbt documentation and keep metrics in sync with downstream dashboards (whether they are checked into version control or not). The Glean dbt integration can also be integrated in CI/CD workflows.
```

<Steps>

### Configure your `dbt_project.yml`

You can use the `glean-defaults` key to specify the glean version (which is required) and connection information.

```yaml filename="dbt_project.yml" copy

models:
  +meta:
    # glean-defaults settings will be merged into each Glean model
    glean-defaults:
      glean: "1.0"

>       # preventUpdatesFromUI defaults to false, but can be set to true

Reviewer: `preventUpdatesFromUI` defaults to `true`.

      preventUpdatesFromUI: false
      source:
        connectionName: Uploads # the name of the conneciton in Glean
        # can use jinja
        # schema is always optional

>         schema: "{{ env_var('DBT_SCHEMA') }}"

Reviewer: would maybe suggest using `"{{ target.schema }}"` which might be more familiar to dbt users

```

### Define model information

The simplest example of a Glean model just defines a metric.  Glean models must include the `glean` key in the dbt model.

```yaml filename="models/schema.yml" copy
version: 2

models:
  # 
  - name: order_attribution
    description: This model describes how each order is attributed to different marketing touchpoints.
    columns:
      - name: attribution_id
        description: Unique ID for the attribution.
      - name: order_id
        description: Order ID, matches the order_id in the sales model.
      - name: touchpoint_id
        description: ID of the marketing touchpoint which led to the order.

>     config:

Reviewer: `config` not needed
